File: api/app/models/comment.rb
class Comment < ActiveRecord::Base
  acts_as_readable on: :created_at

  belongs_to :commentable, polymorphic: true
  belongs_to :profile
  belongs_to :team # team for profile in deadline (if exists)
  belongs_to :trade_deadline

  validates_presence_of :body, :commentable, :profile

  before_save :set_cached_values

  private

  def set_cached_values
    if commentable.is_a? TradeDeadline
      self.trade_deadline = commentable
    elsif commentable.respond_to? :trade_deadline
      self.trade_deadline = commentable.trade_deadline
    end

    if trade_deadline
      self.team = trade_deadline.team_for_profile(profile)
    end
  end
end

File: web/app/models/comment.js
import DS from 'ember-data';
import Readable from 'hitgrid/mixins/readable';

const { Model, attr, belongsTo } = DS;

export default Model.extend(Readable, {
  body: attr(),
  commentable: belongsTo('commentable', { polymorphic: true }),
  profile: belongsTo('profile', { async: false }),

  team: belongsTo('team', { async: false }),
  tradeDeadline: belongsTo('tradeDeadline', { async: false }),

  createdAt: attr('moment')
});
